src/goto-symex/symex_atomic_section.cpp

```diff
@@ -36,8 +35,7 @@ void goto_symext::symex_atomic_end(statet &state)
   if(state.guard.is_false())
     return;
 
-  if(state.atomic_section_id==0)
-    throw "ATOMIC_END unmatched"; // NOLINT(readability/throw)
+  INVARIANT(state.atomic_section_id != 0, "ATOMIC_END unmatched");
```

Comment:
Also an exception, as you can easily write a program that does not have matching begin/end pairs.